src/complex/Parameters/FileSystemPathParameter.cpp

```diff
@@ -47,9 +59,32 @@ Result<> ValidateInputDir(const FileSystemPathParameter::ValueType& path)
   return {};
 }
 
+//-----------------------------------------------------------------------------
+Result<> ValidateDirectoryWritePermission(const FileSystemPathParameter::ValueType& path)
+{
+  auto checkedPath = path;
+  while(!fs::exists(checkedPath))
+  {
+    checkedPath = checkedPath.parent_path();
+  }
+  // We should be at the top of the tree with an existing directory.
+  int accessPerms = FSPP_ACCESS(checkedPath.string().c_str(), FSPP_CHECK_WRITABLE);
```

Comment:
I'd wrap this in it's own function. It'll make it easy to extract if we need it somewhere else in the future.
```cpp
bool HasWriteAccess(const std::string& path)
{
  return FSPP_ACCESS(path.c_str(), FSPP_CHECK_WRITABLE) == 0;
}
```